pkcs11/src/new/mod.rs

```diff
@@ -46,25 +43,46 @@ impl Pkcs11 {
 pub enum Error {
     /// Any error that happens during library loading of the PKCS#11 module is encompassed under
     /// this error. It is a direct forward of the underlying error from libloading.
-    LibraryLoading { err: libloading::Error },
+    LibraryLoading(libloading::Error),
 
     /// All PKCS#11 functions that return non-zero translate to this error. Note though that only true
     /// errors will be returned as such. Some functions that return non-zero values that are not errors
     /// will not be returned as errors. The affected functions are:
     /// `get_attribute_value`, `get_function_status`, `cancel_function` and `wait_for_slot_event`
-    Pkcs11(types::function::Rv),
+    Pkcs11(types::function::RvError),
+
+    /// This error marks a feature that is not yet supported by the PKCS11 Rust abstraction layer.
+    NotSupported,
```

Comment:
Reckon we'll need more wrapper-layer-specific error codes?

Reply:
I think so, I have added a couple more for object management as well..

Reply:
Are you just throwing them in this enum or nesting them?

Reply:
Ah I see what you mean, well I have just throwed them in the enum for now, we can create a specific type after!